cranelift/isle/isle/Cargo.toml

```diff
@@ -11,12 +11,14 @@ version = "0.89.0"
 [dependencies]
 log = { version = "0.4", optional = true }
 miette = { version = "5.1.0", optional = true }
+rayon = "^1.5"
```

Comment:
This adds a lot of dependencies to Cranelift. Is the overlap checking really so expensive that paralleling it makes a lot of difference?

Reply:
I just checked it. Takes less than half a second total even when not parallelizing. Will open a PR to remove rayon.

Reply:
We depend on rayon elsewhere in cranelift, which is why we added it to the overlap checker. Were you removing it completely from cranelift?

Reply:
wasmtime-cranelift depends on rayon, but cranelift itself doesn't apart from cranelift-isle because of this PR. https://github.com/bjorn3/rustc_codegen_cranelift/commit/266e96785ab71834b917bf474f130a6d8fdecd4b reverted cg_clif back to cranelift 0.88.1 from before this PR. This commit removes rayon from cg_clif's Cargo.lock.

Reply:
Ah I see. My assumption had been that since it was in `cranelift/Cargo.toml` that it would be okay to rely on it in isle.

Reply:
FWIW `cranelift/Cargo.toml` defines the crate for `cranelift-tools`, the CLI utility; we do want to keep dependencies of `cranelift-codegen` as minimal as possible, and since `cranelift-isle` is a build-dep of that I'd consider the same policy to apply. Sorry for not catching this earlier!

Reply:
If I remember correctly, when we added the rayon dependency, overlap checking was dramatically slower, so parallelism shaved a lot of time off the wasmtime build. But by the time we actually merged overlap checking to `main`, we'd fixed its performance, so rayon didn't matter any more; we just didn't re-evaluate it at that point.

So thank you for #5101: it's a good change.